### PR TITLE
AC-1564 Added Source UI and fixed existing tests.

### PR DESCRIPTION
**Pull request by woodheadio**
### Description

Implements the UI for the Sources Management tab on the Configuration page.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New unit tests added to cover the changes.
- [ ] New e2e tests added to cover the changes. (If it has been decided these will not be added with the PR, a separate ticket **must** be created and linked in the ticket below before merge)
- [x] All tests passed locally.

### CODE REVIEW

File: src/components/sources-configuration/ConfigurationModal.vue [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        ...this.requiredTextRules,
+        (value: string) =>
+            /^[a-zA-Z0-9_]*$/.test(value) || "Only alphanumeric and underscores are allowed",
+    ];
+    aeTitleCharactersRules: InputValidationRules = [
+        ...this.requiredTextRules,
+        (value: string) =>
+            /^[^\\+^\s]*([^\\]+\s)*[^\\+\s]+$/.test(value) ||
+            "AE Title cannot contain backslash, leading or trailing spaces",
+    ];
```
Comment (samrooke) [inline src/components/sources-configuration/ConfigurationModal.vue:196]: I wonder if it's worth having the regex methods in a utils method maybe? I think we already have similar it not the same validation happening in destinations so it's duplicate effort here.

File: src/components/sources-configuration/ConfigurationModal.vue [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            },
+        },
+        port: {
+            get() {
+                return this.$props.source?.port;
+            },
+            set(value: string) {
+                this.$data.sourceDetails.port = value;
+            },
+        },
```
[the remark this answers is not in the document]
Comment (woodheadio) [reply src/components/sources-configuration/ConfigurationModal.vue:145]: Good spot. We missed that on the [ticket for this work](https://answerconsulting.jira.com/browse/AC-1564)

File: src/components/sources-configuration/ConfigurationModal.vue (hunk 1 above)
Comment (rowellx68) [reply src/components/sources-configuration/ConfigurationModal.vue:196]: the regex that's used by the backend validation is this. it uses [validator](https://www.npmjs.com/package/validator) underneath. maybe we could just pull that library in?

```
((?:[0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])[.]){3}(?:[0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])
```

File: src/components/sources-configuration/ConfigurationModal.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (woodheadio) [reply src/components/sources-configuration/ConfigurationModal.vue]: Removed Port

**Review by samrooke — APPROVED** [review]

**Comment by woodheadio** [comment]
Waiting on the backend change to go in.

**Pull request merged**